storage/storage-client/src/lib.rs

```diff
@@ -54,6 +55,63 @@ use storage_proto::{
 };
 use tokio::runtime::Runtime;
 
+pub struct StorageServiceClientSimple {
+    network_client: Mutex<NetworkClient>,
+}
+
+impl StorageServiceClientSimple {
+    pub fn new(server_address: &SocketAddr) -> Self {
+        Self {
+            network_client: Mutex::new(NetworkClient::new(*server_address)),
+        }
+    }
+
+    fn request(
```

Comment:
You know, the funny thing is, when folks see their own coding style used elsewhere, they can recognize issues with it. I wonder if we cannot make this function request<T: DeserializedOwned> -> Result<T, ...>
That way you perform the deserialization / error checking in place. And your input output function only sees meaningful types.